### ISSUE TITLE
Classifier: `validate_err` scales as 1/n, `bias_err` does not

### ISSUE THREAD

**Issue by HDZucht** [kvoyager/torchsonn#1]
<p><strong>Location:</strong> <code>gmdhpy/neuron.py:190-197, 243-257</code></p>
<pre><code class="language-python">if loss == "mse":          self.loss_function = self._mse;  self.loss_norm = self._mse_norm
elif loss == "logloss":    self.loss_function = log_loss;   self.loss_norm = self._logloss_norm

def _mse(self, y, yp):       return ((y - yp) ** 2).sum()     # SUM
def _mse_norm(self, y):      return (y ** 2).sum()            # SUM    -> ratio is O(1)  ✓
def _logloss_norm(self, y):  return np.absolute(y).sum()      # SUM
# but sklearn.metrics.log_loss returns the MEAN                        -> ratio is O(1/n) ✗
</code></pre>
<p>The regressor divides a sum by a sum. The classifier divides a mean by a sum, so <code>validate_err</code> carries sample size as a factor.</p>
<p><strong>Reproduction.</strong> Take one dataset and duplicate it k times. The distribution is unchanged, the fitted model is unchanged, only n changes. Anything that moves is arithmetic, not statistics.</p>

k | n | Regressor validate_err | Classifier validate_err
-- | -- | -- | --
1 | 300 | 2.9201e-01 | 4.8112e-03
2 | 600 | 2.9171e-01 | 2.3811e-03 (2.02× smaller)
4 | 1200 | 2.9156e-01 | 1.1844e-03 (4.06× smaller)
8 | 2400 | 2.9149e-01 | 5.9061e-04 (8.15× smaller)

<p>The regressor stays constant, as it should. The classifier falls by exactly k. <code>bias_err</code> stays constant in both cases (6.9e-03 → 7.6e-03 across the same range).</p>
<p><strong>Consequence.</strong> <code>CriterionType.cmpComb_validate_bias</code> mixes <code>0.5 * bias_err + 0.5 * validate_err</code> (<code>neuron.py:117-118</code>). If only one of the two terms scales with 1/n, the mixture depends on dataset size: at n = 300 the terms are comparable (4.8e-03 vs 6.9e-03), at n = 2400 the bias term dominates by more than tenfold. On large datasets <code>validate_bias</code> is effectively a pure bias criterion, and nothing in the API signals this.</p>
<p>Plain <code>validate</code> selection is unaffected: within one run the denominator is constant, so the ranking holds.</p>
<p><strong>Suggested fix.</strong> Normalize the log loss by the cross-entropy of the base-rate model, the classification analogue of the regularity criterion's zero predictor:</p>
<pre><code class="language-python">def _logloss_norm(self, y):
    p = np.clip(y.mean(), 1e-12, 1 - 1e-12)
    return -(p * np.log(p) + (1 - p) * np.log(1 - p))   # base-rate entropy, size-independent
</code></pre>
<p>The error then reads ≈ 1 for a useless model and ≈ 0 for a perfect one, matching the regression path, and both criteria live on the same scale again.</p>

**Comment by kvoyager** [comment]
This doesn't apply to torchsonn, both criterion pairs are already consistently scaled:
- Multi-class — regularity is sum_nll / total_N / H_Y (trainer.py:1560); bias is js_sum / (N · ln2) (errors.py:308) under the default bias_ce_type='js', or Σ(z_A-z_B)²/N under 'l2'. Per-sample either way, so the error_alpha mix at model.py:544 is size-stable — and the weight is a config field, not a hardcoded 0.5. 
- Regression/binary — regularity and bias are both Σ/Σ and both take the same centered flag from model.error_centered (trainer.py:1584, trainer.py:1639). errors.py: 79-82 states that invariant explicitly as the reason.
- Binary goes through the MSE branch, not a log loss, so the mean/sum mismatch has no way to arise.
